src/com/company/EchoServer.java

```diff
@@ -0,0 +1,51 @@
+package com.company;
+
+import java.io.IOException;
+import java.net.ServerSocket;
+import java.net.Socket;
+
+public class EchoServer {
+
+    private ServerSocket serverSocket;
+    private int portNumber;
+    private ServerMessenger serverMessenger;
+
+    public EchoServer(ServerSocket serverSocket, int portNumber, ServerMessenger serverMessenger) {
+        this.serverSocket = serverSocket;
+        this.portNumber = portNumber;
+        this.serverMessenger = serverMessenger;
+    }
+
+    public void run() {
+        serverMessenger.printServerPort(portNumber);
+        while (runServer()) {
+            connectWithSocket();
+        }
+    }
+
+    public boolean runServer() {
+        return true;
+    }
+
+    private void connectWithSocket() {
+        try {
+            Socket clientSocket = accept();
+            serverMessenger.informOfNewSocket();
+            echo(clientSocket);
+        } catch (IOException exception) {
+            serverMessenger.informOfException(this.portNumber, exception.getMessage());
+        }
+    }
+
+    private Socket accept() throws IOException {
```

Comment:
You could get rid of this method seeing as it only contains one line and that line is already as informative as the method name `accept()` anyways.  Unless this method name gives any more information to the reader or has a few lines of related code, then it is unnecessary.

Reply:
done.